core/src/message/context.rs

```diff
@@ -278,21 +278,22 @@ impl MessageContext {
         packet: InitPacket,
         delay: u32,
     ) -> Result<(MessageId, ProgramId), Error> {
-        let program_id = packet.destination();
-
-        if self.store.initialized.contains(&program_id) {
-            return Err(Error::DuplicateInit);
-        }
-
         let last = self.store.outgoing.len() as u32;
 
         if last >= self.settings.outgoing_limit {
             return Err(Error::OutgoingMessagesAmountLimitExceeded);
         }
 
         let message_id = MessageId::generate_outgoing(self.current.id(), last);
+
         let message = InitMessage::from_packet(message_id, packet);
```

Comment:
wrong id is used to create a program id

Reply:
What should be used then? self.current.id()? that seems like it would generate duplicates.